Code/Framework/AzCore/AzCore/std/string/conversions.h

```diff
@@ -337,7 +337,11 @@ namespace AZStd
 
         if (srcLen > 0)
         {
-            Internal::WCharTPlatformConverter<>::to_string(dest, destSize, str, str + srcLen + 1); // copy null terminator
+            char* endStr = Internal::WCharTPlatformConverter<>::to_string(dest, destSize, str, str + srcLen);
+            if (*(endStr - 1) != '\0')
+            {
+                *endStr = '\0'; // copy null terminator
+            }
```

Comment:
There shouldn't be a need to check that the endStr previous character.
The source string that was converted over is actualy string porition without the null-terminating character.

All that should actually checked is if the `endStr` is less than the `dest + destSize`

Reply:
`srcLen` can be passed and in that case it could be a  non-null-terminated string
I agree that I should also check that `endWStr <= (dest + destSize + 1)` to not write outside of the buffer

Reply:
My point about the `endStr` check is if the last character before `endStr`  is NUL-Terminating character, then it is still part of the converted "string".
It just means that the user supplied a a non-null-terminated string that contained a NUL-Terminating Character within it.

i.e
`AZStd::wstring testString(L"My\0Foo\0");`

The actual string is 7 characters and should result in a destination that is `{'M', 'y', 0, 'F', 'o', 'o' 0, 0}`

The caller should be able to determine the size of the converted string if they wanted to store it in an `AZStd::string` later

Reply:
talked to @lumberyard-employee-dm offline and he was pointing out that the check `*(endStr - 1) != '\0'` wasnt needed because in the 3 cases it would be null-terminated:
- if `srcLen` is not passed, then the `\0` is not copied to `dest`, therefore it should be set
- if `srcLen` is passed with the length of the C-style string, then the `\0` is not copied to `dest`, therefore it should be set
- if `srcLen` is passed accounting for the `\0` (length of string + 1) then it `\0` will be copied, in this case, it will add an extra `\0` at the end
In all cases, the check `(endWStr < (dest + destSize))` prevents to overflow the buffer

So, removed the  `*(endStr - 1) != '\0'` check.